cluster/state.go

```diff
@@ -16,10 +17,10 @@ type state struct {
 	Nodes      []common.Node
 }
 
-// TODO: this should be replaced by a configurable path later
-var statePath = "/var/lib/wesher/state.json"
+var defaultStatePath = "/var/lib/wesher/%s.json"
```

Comment:
Changing the state file name has the unfortunate consequence of breaking existing clusters on upgrade. 

I think we have to add some special case handling for the "default" cluster and also try to load the old state file (`state.json`) if the new one (`default.json`) doesn't exist. Ugly, but rather this than breaking people's clusters.
And just for completeness' sake: calling the default cluster "state" feels even uglier :wink:

Reply:
Waiting for the other comment thread to be resolved since it will impact this as well.